getting_started_modules.md

```diff
@@ -0,0 +1,332 @@
+  debug:
+    msg: "{{terraform.stdout}}"
+```
+
+
+#### 5. Making The Nomad Job More Dynamic With Terraform Variables
+At this point we have a "hardcoded" nomad job. 
+In the example there isn't really many variables that will benefit from being more dynamic, but we'll demonstrate using the image-name in this case. 
+We want to use this to let the user choose their own image in our module. Create a `variables.tf` in the root-directory, then add the lines below
+
+```hcl-terraform
+variable "docker_image" {
+  type        = string
+  description = "docker image to use in module"
+}
+```
+ 
+Now try to run this module like you've done earlier. 
+At this point you'll get an error TODO:ADD ACTUAL ERROR MSG. 
+This is because there is no code anywhere defining `docker_image `. 
+
+How dynamic the nomad job suppose to be is defined by the needs of the end-user who are going to use this module. There are two main methods that we recommend.
+
+##### Method 1 - extract dynamic data into variables.tf
+
+We advise starting simple with doing extraction of potentially dynamic data in nomad job, such as:
+- job name
+- list of data centers
+- namespace (enterprise only)
+- properties of [update stanza](https://www.nomadproject.io/docs/job-specification/update)
+- [service name stanza](https://www.nomadproject.io/docs/job-specification/service#name)
+- env variables in [template stanza](https://www.nomadproject.io/docs/job-specification/template)
+- docker image in [task -> config -> image](https://www.nomadproject.io/docs/drivers/docker)
+- ...
+
+> Extracting variables is a process, so you do not need to extract all dynamic data at once. Try to prioritize user needs.
+
+##### Method 2 - conditional rendering of nomad job
+You can add a variable in `variables.tf` to define a condition.
+This variable will be used to control if-statement how to render the nomad job.
+
+Example:  
+
+`variables.tf`
+
+/```hcl
+variable "use_canary" {
+  type = bool
+  description = "Uses canary deployment for Presto"
+  default = false
+}
+/```
+
+`nomad-job.hcl`
+/```hcl
+...
+update {
+  max_parallel      = 1
+  health_check      = "checks"
+  min_healthy_time  = "10s"
+  healthy_deadline  = "12m"
+  progress_deadline = "15m"
+  %{ if use_canary }
+  canary            = 1
+  auto_promote      = true
+  auto_revert       = true
+  %{ endif }
+  stagger           = "30s"
+}
+...
+/```
+
+
+#### 6. Integrating The Nomad Job With Vault
+When we create modules we often need to provide our service with a key or a secret, and in some cases both. 
+However, in either case we can use [Vault](https://www.vaultproject.io/) to store keys and secrets that we can get later in our code. 
+> :warning: All communication between `terraform` and `vault` is handled by [terraform-provider-vault](https://registry.terraform.io/providers/hashicorp/vault/latest/docs)
+Let's use [Postgres](https://www.postgresql.org/) as an example, and say we need to provide it with a username and password. 
```

Comment:
```suggestion
> :warning: All communication between `terraform` and `vault` is handled by [terraform-provider-vault](https://registry.terraform.io/providers/hashicorp/vault/latest/docs)

Let's use [Postgres](https://www.postgresql.org/) as an example, and say we need to provide it with a username and password. 
```

Reply:
Github parsed this as a quote so need to add spacing

Reply:
Thanks!